main.tf

```diff
@@ -4,22 +4,66 @@ resource "random_integer" "this" {
 }
 
 locals {
-  azureopenai_account_name         = coalesce(var.azureopenai_account_name, "azureopenai-${random_integer.this.result}")
-  azureopenai_customsubdomain_name = coalesce(var.azureopenai_customsubdomain_name, "azureopenai-${random_integer.this.result}")
+  account_name          = coalesce(var.account_name, "azure-openai-${random_integer.this.result}")
+  custom_subdomain_name = coalesce(var.custom_subdomain_name, "azure-openai-${random_integer.this.result}")
   tags = var.default_tags_enabled ? {
     Application_Name = var.application_name
     Environment      = var.environment
   } : {}
 }
 
 resource "azurerm_cognitive_account" "this" {
-  kind                          = "OpenAI"
-  location                      = local.location
-  name                          = local.azureopenai_account_name
-  resource_group_name           = data.azurerm_resource_group.this.name
-  sku_name                      = var.sku_name
-  custom_subdomain_name         = local.azureopenai_customsubdomain_name
-  public_network_access_enabled = var.public_network_access_enabled
+  kind                               = "OpenAI"
+  location                           = local.location
+  name                               = local.account_name
+  resource_group_name                = data.azurerm_resource_group.this.name
+  sku_name                           = var.sku_name
+  custom_subdomain_name              = local.custom_subdomain_name
+  public_network_access_enabled      = var.public_network_access_enabled
+  dynamic_throttling_enabled         = var.dynamic_throttling_enabled
+  fqdns                              = var.fqdns
+  local_auth_enabled                 = var.local_auth_enabled
+  outbound_network_access_restricted = var.outbound_network_access_restricted
+
+  dynamic "customer_managed_key" {
+    for_each = var.customer_managed_key != null ? [var.customer_managed_key] : []
+    content {
+      key_vault_key_id   = customer_managed_key.value.key_vault_key_id
+      identity_client_id = customer_managed_key.value.identity_client_id
+    }
+  }
+
+  dynamic "identity" {
+    for_each = var.identity != null ? [var.identity] : []
+    content {
+      type         = identity.value.identity_type
+      identity_ids = identity.value.identity_ids
+    }
+  }
+
+  dynamic "network_acls" {
+    for_each = var.network_acls != null ? [var.network_acls] : []
+    content {
+      default_action = network_acls.value.default_action
+      ip_rules       = network_acls.value.ip_rules
+      dynamic "virtual_network_rules" {
+        for_each = network_acls.value.virtual_network_rules != null ? [network_acls.value.virtual_network_rules] : []
```

Comment:
According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.56.0/internal/services/cognitive/cognitive_account_resource.go#L221-L225) this resource support multiple `virtual_network_rules`, so the `for_each` should be:

```hcl
for_each = network_acls.value.virtual_network_rules
```